funnel/models/membership.py

```diff
@@ -137,7 +147,9 @@ def replace(self, actor, **roles):
             raise AttributeError("Unknown role")
         self.revoked_at = db.func.utcnow()
         self.revoked_by = actor
-        new = type(self)(user=self.user, parent=self.parent, granted_by=self.granted_by)
+        new = type(self)(
+            user=self.user, parent_id=self.parent_id, granted_by=self.granted_by
+        )
```

Comment:
Why the switch to `parent_id`?

Reply:
Because `parent_id` can be `None`. so `self.parent` might raise an error I thought. Havent tested that yet. could that be an issue?